src/main/java/io/servertap/PluginEntrypoint.java

```diff
@@ -5,18 +5,43 @@
 import io.servertap.api.v1.PlayerApi;
 import io.servertap.api.v1.ServerApi;
 import org.bukkit.plugin.java.JavaPlugin;
+import java.util.logging.Logger;
+
+import net.milkbowl.vault.chat.Chat;
+import net.milkbowl.vault.economy.Economy;
+import net.milkbowl.vault.economy.EconomyResponse;
+import net.milkbowl.vault.permission.Permission;
+
+import org.bukkit.Bukkit;
+import org.bukkit.command.Command;
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Player;
+import org.bukkit.plugin.RegisteredServiceProvider;
+import org.bukkit.plugin.java.JavaPlugin;
 
 import java.util.logging.Logger;
 
 import static io.javalin.apibuilder.ApiBuilder.*;
 
 public class PluginEntrypoint extends JavaPlugin {
 
-    private final Logger log = getLogger();
+    private static final Logger log = Bukkit.getLogger();
+
+
+    private static Economy econ = null;
+
+
+
 
     @Override
     public void onEnable() {
 
+        if (!setupEconomy() ) {
+            log.severe(String.format("[%s] - Disabled due to no Vault dependency found!", getDescription().getName()));
+            getServer().getPluginManager().disablePlugin(this);
```

Comment:
I think the way this is currently implemented means that this plugin won't work at all if Vault is not available on the server. Are you able to re-work these changes so that it just sets a `hasEconomy` bool, and then any of the API methods that interact with Vault would just return an error `no economy available` or something.

I think this way, you can still use servertap without needing Vault if you want.

Reply:
I was thinking the same thing as I was working on this. I have a fairly good idea of how to make it a soft dependency and will try implementing soon. I'll ping you when I get it working.